kv/kvserver: TestStoreRangeMergeConcurrentRequests failed

kv/kvserver.TestStoreRangeMergeConcurrentRequests [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18039960?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18039960?buildTab=artifacts#/) on master @ [258fe7d2f9eeb8d2d653167fe815126080c2eb56](https://github.com/cockroachdb/cockroach/commits/258fe7d2f9eeb8d2d653167fe815126080c2eb56):


```
=== RUN   TestStoreRangeMergeConcurrentRequests
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/d351c1750c446bc1442d58db9c0ceef9/logTestStoreRangeMergeConcurrentRequests3362232039
    test_log_scope.go:76: use -show-logs to present logs inline
    client_merge_test.go:2378: context deadline exceeded
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d351c1750c446bc1442d58db9c0ceef9/logTestStoreRangeMergeConcurrentRequests3362232039
--- FAIL: TestStoreRangeMergeConcurrentRequests (52.10s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45241

kv/kvserver.TestStoreRangeMergeConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/a3b78e3b-0ad5-495b-ad9a-a9e70ce6adfa?testReportRun=14&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [fa9acaa8a8b8b58b1448c61ca5a8f6e6a992d64c](https://github.com/cockroachdb/cockroach/commits/fa9acaa8a8b8b58b1448c61ca5a8f6e6a992d64c):


```
=== RUN   TestStoreRangeMergeConcurrentRequests
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestStoreRangeMergeConcurrentRequests449201686
    test_log_scope.go:76: use -show-logs to present logs inline
    client_merge_test.go:2378: context deadline exceeded
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestStoreRangeMergeConcurrentRequests449201686
--- FAIL: TestStoreRangeMergeConcurrentRequests (57.59s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=14</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -2331,9 +2331,10 @@ func TestStoreRangeMergeConcurrentRequests(t *testing.T) {
 		}
 	}
 
-	// Failures in this test often present as a deadlock. Set a short timeout to
-	// limit the damage, but a longer deadline for race builds.
-	ctx, cancel := context.WithTimeout(ctx, testutils.SucceedsSoonDuration())
+	// Failures in this test often present as a deadlock.
+	// We have a relatively high timeout since this test flakes in leader leases,
+	// as it keeps withdrawing/granting store liveness support.
+	ctx, cancel := context.WithTimeout(ctx, 3*testutils.DefaultSucceedsSoonDuration)
 	defer cancel()
 
 	const numGetWorkers = 16

